.github/workflows/pre-commit.yaml

```diff
@@ -37,7 +37,7 @@ jobs:
         uses: actions/cache@v2
         with:
           path: ~/.cache/pre-commit
-          key: ${{ runner.os }}-${{ hashFiles('.pre-commit-config.yaml') }}
+          key: ${{ matrix.os }}-${{ matrix.python-version }}-${{ hashFiles('requirements.txt') }}
```

Comment:
the hash of pre-commit test should be calculated from `.pre-commit-config.yaml` rather than `requirements.txt`